docs/generating_tasks.md

```diff
@@ -157,6 +157,49 @@ of the task definition. When this tag is present, both the extra setup steps and
 of multiversion sub-tasks will be preformed. In order to only perform the extra setup steps
 the `"no_version_combinations"` tag should also be included.
 
+### Burn in tests and burn in tags
+
+Newly added or modified tests might become flaky. In order to avoid that, those tests can be run
+continuously multiple times in a row to see, if the results are consistent. This process is called
+burn-in.
+
+`burn_in_tests_gen` task is used to generate burn-in tasks on the same buildvariant the task is
+added to. [Example](https://github.com/mongodb/mongo/blob/81c41bdfdc56f05973fae70e80e80919f18f50c9/etc/evergreen_yml_components/definitions.yml#L3252-L3256)
+of task configuration:
+
+```yaml
+- <<: *gen_task_template
+  name: burn_in_tests_gen
+  tags: []
+  commands:
+  - func: "generate resmoke tasks"
+```
+
+`burn_in_tags_gen` task is used to generate separate burn-in buildvariants. This way we can add the
+task to the required buildvariant, that will generate another required buildvariants on the fly
+based on another non-required buildvariants with only necessary burn-in tasks.
```

Comment:
[nit] This way we can burn_in on the the requested buildvariant as well other, additional variants to ensure there is no difference between variants.

Reply:
updated